src/main/kotlin/Main.kt

```diff
@@ -41,4 +58,27 @@ fun problem2(charArray: Array<Char>) : String{
         }
     }
     return "Not a single letter of the alphabet is missing in this array!"
-}
\ No newline at end of file
+}
+
+// Jenny and Stephanie are friends. Both of them like letters, especially uppercase ones.
+// They are cutting uppercase letters from newspapers, and each one of them has his collection of letters stored in a stack.
+// One sunny day, Rachel visited Jenny and Stephanie. She saw their collections. She wondered what is the lexicographically minimal string made of those two collections. She can take a letter from a collection only when it is on the top of the stack. Rachel wants to use all the letters in their collections.
+// As an example, assume Jenny has collected a = [A, C, A] and Stephanie has b = [B, C, F]. The example shows the top at index 0 for each stack of letters. Assemble the string as follows:
+fun problem3(jenny: LinkedList<Char>, stephanie: LinkedList<Char>) : LinkedList<Char>{
```

Comment:
Why did you choose linked list for this?

Reply:
Because if I had to use... let's say a Array List it would be less efficient because my lists will change in size with time so a more faster aproach in my opinion is a LinkList where deleting or adding of a object will not need to copy the original content + modifications such as shrinking or getting bigger... with link list it's simpler and much more efficient in this case where you need to change sizes more often because of pop and add methods you sill only link a pointer to next value stored or you will remove that value with the pointer to that value so only complexity would be smaller then in Array List case where if you have a n sized list you will need to copy n values and add one value for adds or just copy n - 1 values for pop to get the new list.